### PR TITLE
Invites: dispatch error notice on api error

### PR DESCRIPTION
**Pull request by lezama**
Alternative to https://github.com/Automattic/wp-calypso/pull/2595

### CODE REVIEW

File: client/lib/invites/actions.js [hunk 1]
```diff
@@ -71,29 +72,35 @@ export function createAccount( userData, callback ) {
...
+				} );
 
-			if ( error ) {
-				analytics.tracks.recordEvent( 'calypso_invite_accept_failed' );
-			} else {
-				analytics.tracks.recordEvent( 'calypso_invite_accepted' );
-			}
-			if ( typeof callback === 'function' ) {
-				callback( error, data );
+				if ( error ) {
```
Comment (ebinnion) [inline client/lib/invites/actions.js]: We should probably modify this to be `if ( error && error.message )`

**Comment by ebinnion** [comment]
I noticed that for logged out invites, if the accept invite request fails, that the invite gets stuck. Perhaps we should redirect to the reader in this case?

screen shot

File: client/lib/invites/actions.js [hunk 2]
```diff
@@ -55,45 +57,56 @@ export function fetchInvite( siteId, inviteKey ) {
...
+		Dispatcher.handleViewAction( {
+			type: ActionTypes.INVITE_ACCEPTED,
+			invite,
+			displayOnNextPage: true
+		} );
+		wpcom.undocumented().acceptInvite(
+			invite,
+			( error, data ) => {
+				Dispatcher.handleViewAction( {
+					type: error ? ActionTypes.INVITE_ACCEPTED_ERROR : ActionTypes.INVITE_ACCEPTED_SUCCESS,
```
Comment (roccotripaldi) [inline client/lib/invites/actions.js]: RECEIVE_INVITE_ACCEPTED_ERROR, RECEIVE_INVITE_ACCEPTED_SUCCESS

File: client/my-sites/invites/controller.js [hunk 3]
```diff
@@ -14,10 +14,16 @@ import titleActions from 'lib/screen-title/actions';
...
 import { renderWithReduxStore } from 'lib/react-helpers';
-import { acceptedNotice, getRedirectAfterAccept } from 'my-sites/invites/utils';
-import { successNotice } from 'state/notices/actions';
+import { getRedirectAfterAccept } from 'my-sites/invites/utils';
+import { acceptInvite as acceptInviteAction } from 'lib/invites/actions';
 
 export function acceptInvite( context ) {
+	const acceptedInvite = store.get( 'invite_accepted' );
+	if ( acceptedInvite && acceptedInvite.loggedOut ) {
+		acceptInviteAction( acceptedInvite, page.bind( null, getRedirectAfterAccept( acceptedInvite ) ) )( context.store.dispatch );
```
Comment (ebinnion) [inline client/my-sites/invites/controller.js]: We probably need to clear the `invite_accepted` from `localStorage` within this condition.

```
store.remove( 'invite_accepted' );
```

File: client/lib/user/index.js [hunk 4]
```diff
@@ -31,6 +31,11 @@ User.dispatchToken = Dispatcher.register( function( payload ) {
 				incrementSiteCount();
 			}
 			break;
+		case InvitesActionTypes.INVITE_ACCEPTED_ERROR:
```
Comment (lezama) [inline client/lib/user/index.js]: we can remove this

**Comment by roccotripaldi** [comment]
This is pretty great. I tested in a handful of scenarios, and the notification flow is very much improved, everything is working well, and there are no errors. ( Needs a rebase ).

I do not fully grok what is happening with `bindActionCreators` and `connect`, but i'd say it is ready to ship after the rebase.

File: client/my-sites/invites/controller.js [hunk 5]
```diff
@@ -14,10 +14,24 @@ import titleActions from 'lib/screen-title/actions';
...
+import { getRedirectAfterAccept } from 'my-sites/invites/utils';
+import { acceptInvite as acceptInviteAction } from 'lib/invites/actions';
 
 export function acceptInvite( context ) {
+	const acceptedInvite = store.get( 'invite_accepted' );
+	if ( acceptedInvite ) {
+		store.remove( 'invite_accepted' );
+		const acceptInviteCallback = error => {
+			if ( error ) {
+				page( window.location.href );
```
Comment (ebinnion) [inline client/my-sites/invites/controller.js]: I'm not sure that we should be redirecting right here. After redirecting, we are no longer showing the error notice and instead show the logged in invite accept.

File: client/lib/sites-list/list.js [hunk 6]
```diff
@@ -187,6 +190,7 @@ SitesList.prototype.parse = function( data ) {
  * Merge changes to existing sites and remove any sites that are not present
  **/
 SitesList.prototype.update = function( sites ) {
+	console.log( 'update sites:', sites )
```
Comment (ebinnion) [inline client/lib/sites-list/list.js]: :smile:

File: client/my-sites/invites/controller.js (hunk 5 above)
Comment (ebinnion) [inline client/my-sites/invites/controller.js]: For what it's worth, I think the biggest flow issue is due to the force matching email issue, and once we get in #2647, this should be much better. So, we can probably leave this as is.

**Comment by ebinnion** [comment]
I only saw one issue with the code. Besides that, this tested pretty well.

Let's get the console removed and then :ship:

**Pull request merged**